CodingTracker.Speedierone/CodingTracker.Speedierone/UserInput.cs

```diff
@@ -0,0 +1,194 @@
+        var recordId = GetNumberInput("\n\nPlease enter Id of entry you would like to delete or press 0 to return to main menu");
+        
+
+        using(var connection = new SqliteConnection(connectionString))
+        {
+            connection.Open();
+            var tableCmd = connection.CreateCommand();
+            tableCmd.CommandText = $"DELETE from code_tracker WHERE Id = '{recordId}'";
+
+            int rowCount = tableCmd.ExecuteNonQuery();
+
+            if (rowCount == 0)
+            {
+                Console.WriteLine($"\n\nRecord with Id {recordId} doesn't exist. Please press enter to try again.");
+                Console.ReadLine();
+                DeleteRecord();
+            }
+            Console.WriteLine($"Record Id {recordId} deleted. Press any key to continue.");
+            Console.ReadLine();
+        }
+               
+    }
+    internal static void UpdateRecord()
+    {
+        Console.Clear();
+        ViewRecords();
+
+        string connectionString = ConfigurationManager.AppSettings.Get("dbconnectionString");
+
+        var recordId = GetNumberInput("\nPlease enter Id of record you wish to update or press 0 to go back");
+
+        using (var connection = new SqliteConnection(connectionString))
+        {
+            connection.Open();
+            var checkCmd = connection.CreateCommand();
+            checkCmd.CommandText = $"SELECT EXISTS(SELECT 1 FROM code_tracker WHERE Id = {recordId})";
+            int checkQuery = Convert.ToInt32(checkCmd.ExecuteScalar());
+
+            if (checkQuery == 0)
+            {
+                Console.WriteLine($"\nRecord with Id {recordId} doesn't exists");
+                connection.Close();
+                UpdateRecord();
+            }
+
+            string date = Helpers.GetDate();
+            string startTime = Helpers.GetStartTime();
+            string endTime = Helpers.GetEndTime();
+            string timeSpan = Helpers.CodingTime(startTime, endTime);
+
+            var tableCmd = connection.CreateCommand() ;
+            tableCmd.CommandText = $"UPDATE code_tracker SET Date = '{date}', StartTime = '{startTime}', EndTime'{endTime}', TimeSpan = '{timeSpan}' WHERE Id = {recordId}";
+
+            tableCmd.ExecuteNonQuery();
+            connection.Close();
+        }
+    }
+    internal static void AddTimerRecord()
+    {
+        Console.WriteLine("Press enter to start timer or 0 to return to menu.");
+        var startTimer = Console.ReadLine();
+        if (startTimer == "0") MainMenu.ShowMenu();
+        var timer = new Stopwatch();                  
+        timer.Start();
+        var todayDateTime = DateTime.Now;
+        var startDate = DateOnly.FromDateTime(DateTime.Now);
+        var parsedStartDate = startDate.ToString("dd-MM-yy");
+        var timeStart = TimeOnly.FromDateTime(todayDateTime);
+        var parsedTimeStart = TimeOnly.ParseExact(timeStart.ToString(), "HH:mm", new CultureInfo("en-GB"));
```

Comment:
@Speedierone I'm getting this error when trying to use the timer, not sure it has to do with the culture info (it can be a pain sometimes). Are you able to execute this command without errors?

![image](https://user-images.githubusercontent.com/108711476/236079617-dd9c0124-46b3-4b5e-b806-ae553997295e.png)